AbnerEdgar/PlatziApp
# Implement Basic Network Service Layer and Caching

### Summary
This PR sets up the foundation for networking in the app, including API service configuration, endpoint management, and a basic caching system.

### Changes
- Created `APIService` to handle all network requests.
- Added `APIEndpoint` protocols and implementations for organized endpoint management.
- Integrated `HTTPMethod` enum for type-safe HTTP methods.
- Implemented `ProductEndpoint` for handling product-related API requests.

### To-Do
- Fill in missing endpoints for further API integration.
- Add unit tests for network requests.

### How to Test
- Fetch product data using `ProductRepository` and verify the correct response is retrieved.
- Check console logs for successful network requests and responses.
